core/dbt/parser/manifest.py

```diff
@@ -598,6 +600,60 @@ def write_manifest_for_partial_parse(self):
         except Exception:
             raise
 
+    def write_artifacts(self):
+        # write out manifest.json
+        # TODO: check for overlap with parse command writing manifest
+        write_manifest(self.manifest, self.root_project.target_path)
+
+        # build publication metadata
+        metadata = PublicationMetadata(
+            adapter_type=self.root_project.credentials.type,
+            quoting=self.root_project.quoting,
+        )
+
+        # get a list of public model ids first so it can be used in constructing dependencies
+        public_model_ids = []
+        for node in self.manifest.nodes.values():
+            if node.resource_type == NodeType.Model and node.access == AccessType.Public:
+                public_model_ids.append(node.unique_id)
+
+        set_of_public_unique_ids = set(public_model_ids)
+
+        # Get the Graph object from the Compiler
+        from dbt.compilation import Compiler
+
+        compiler = Compiler(self.root_project)
```

Comment:
This is the first time that we're constructing the networkx graph at parse time. I will probably factor out the graph code from the Compiler object a bit better, because it feels wrong to be constructing a Compiler at parse time.

There was some test breakage relating to doing this. Probably we want to skip this for some errors, will have to investigate.

Reply:
Have refactored to put the linker methods in the linker. The test breakage was because of a docs generate test that checked for the existence of manifest.json (it exists now), and because the experimental_parser tests had a model referring to itself, which prior to this caused no error. Now that I'm building the networkx graph we get cycle errors at parse time. Which seems better anyway.